app/pages/lab-pages-editor/components/TasksPage/components/EditStepDialog/types/SingleQuestionTask.jsx

```diff
@@ -0,0 +1,157 @@
+  const [ required, setRequired ] = useState(!!task?.required);
+
+  // Update is usually called manually onBlur, after user input is complete.
+  function update(optionalStateOverrides) { 
+    const _answers = optionalStateOverrides?.answers || answers
+    const nonEmptyAnswers = _answers.filter(({ label }) => label.trim().length > 0);
+
+    const newTask = {
+      ...task,
+      answers: nonEmptyAnswers,
+      help,
+      question,
+      required
+    };
+    updateTask(taskKey, newTask);
+  }
+
+  function addAnswer(e) {
+    const newAnswers = [ ...answers, { label: '', next: undefined }];
+    setAnswers(newAnswers);
+
+    e.preventDefault();
+    return false;
+  }
+
+  function editAnswer(e) {
+    const index = e?.target?.dataset?.index;
+    if (index === undefined || index < 0 || index >= answers.length) return;
+
+    const answer = answers[index];
+    const newLabel = e?.target?.value || '';
+
+    setAnswers(answers.with(index, { ...answer, label: newLabel }));
+  }
+
+  function deleteAnswer(e) {
+    const index = e?.target?.dataset?.index;
+    if (index === undefined || index < 0 || index >= answers.length) return;
+
+    const newAnswers = answers.slice()
+    newAnswers.splice(index, 1);
+    setAnswers(newAnswers);
+    update({ answer: newAnswers });  // Use optional state override, since setAnswers() won't reflect new values in this step of the lifecycle.
+
+    e.preventDefault();
+    return false;
+  }
+
+  // For inputs that don't have onBlur, update triggers automagically.
+  // (You can't call update() in the onChange() right after setStateValue().)
+  useEffect(update, [required]);
+
+  return (
+    <div className="single-question-task">
+      <div className="input-row">
+        <label
+          className="big"
+          htmlFor={`task-${taskKey}-instruction`}
+        >
+          Main Text
+        </label>
+        <div className="flex-row">
+          <span className="task-key">{taskKey}</span>
+          <input
+            className="flex-item"
+            id={`task-${taskKey}-question`}
+            type="text"
+            value={question}
+            onBlur={update}
+            onChange={(e) => { setQuestion(e?.target?.value) }}
+          />
+        </div>
+        {/* <button>Delete</button> */}
+      </div>
+      <div className="input-row">
+        <label className="big">Choices</label>
+        <div className="flex-row">
+          <button
+            aria-label="Add choice"
+            className="big"
+            onClick={addAnswer}
+            type="button"
+          >
+            <PlusIcon />
+          </button>
+          <label className="narrow">
+            <input
+              type="checkbox"
+              checked={required}
+              onChange={(e) => {
+                setRequired(!!e?.target?.checked);
+              }}
+            />
+            <span>
+              Required
+            </span>
+          </label>
```

Comment:
A11y tip: always use `for` with labels, as explicit labels are better supported by assistive tech, eg. Dragon Naturally Speaking.
https://github.com/w3c/aria-practices/issues/2870